src/Controllers/ImageController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace JustBetter\GlideDirective\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Routing\Controller;
+use Illuminate\Support\Str;
+use JustBetter\GlideDirective\Responsive;
+use Statamic\Contracts\Assets\Asset;
+use Statamic\Contracts\Imaging\ImageManipulator;
+use Statamic\Facades\Asset as AssetFacade;
+use Statamic\Facades\Image;
+use Statamic\Imaging\GlideImageManipulator;
+
+class ImageController extends Controller
+{
+    public function getPreset(Request $request, string $preset, string $fit, string $signature, string $file, string $format): Response
+    {
+        $asset = AssetFacade::findByUrl(Str::start($file, '/'));
+
+        if (! $asset) {
+            abort(404);
+        }
+
+        /** @var GlideImageManipulator $manipulator */
+        $manipulator = self::getManipulator($asset, $preset, $fit, $format);
+        $path = $manipulator->build();
```

Comment:
Unfortunately this setup does not guarantee an image is actually generated.
This build is executed on the `UrlBuilder` contract, which changes functionality between builders:
https://github.com/statamic/cms/blob/5.x/src/Providers/GlideServiceProvider.php#L24

Reply:
We want to mimic this behavior https://github.com/statamic/cms/blob/5.x/src/Imaging/StaticUrlBuilder.php#L64